app/assets/javascripts/application.js

```diff
@@ -14,5 +14,7 @@
 //= require jquery_ujs
 //= require turbolinks
 //= require_tree .
+//= require popper
+//= require tether
```

Comment:
We need to require these other libraries _before_ we have the `bootstrap-sprockets` section.  Again, the bootstrap documentation could do a better job of explaining this.